src/tiled/eraser.cpp

```diff
@@ -105,7 +107,10 @@ void Eraser::doErase(bool continuation)
     }
     mLastTilePos = tilePos;
 
-    for (Layer *layer : mapDocument()->selectedLayers()) {
+    auto layers = !mStrong ? mapDocument()->selectedLayers()
+                           : mapDocument()->map()->layers();
```

Comment:
`map()->layers()` only returns the top-level layers for the map, whereas I think you'll want to iterate `map()->tileLayers()`, which also handles group layers (unfortunately that means we need to change the loop a bit since this is not compatibile with `selectedLayers()`. Maybe the loop body could be turned into a lambda.